concepts/modularization/README.md

```diff
@@ -38,17 +39,76 @@ Components can depend only on core Kubernetes API, or on API extensions introduc
 
 If the API you need (like a core Kubernetes API or Istio virtual service) is not available, you should fail. If your component can work without the API, but some features are not available (for example, service monitor from monitoring), you should just skip it and continue to deploy other component resources. 
 
+**Principles:**
+- module operator should know its dependencies
+- keep dependencies minimal and try to depend on API only (check if API exists in the required version)
+- report missing dependencies in the status of the module operator resource
+- remember that dependency can be installed in parallel to your module - give it some time and report error after reasonable timeout
+- error or success is not a final state (your dependencies can come and go) - reconcile, update status, bring it to the desired state eventually
+
 # Release channels
 Release channels let customers try new modules and features early, and decide when the updates should be applied. 
 
-![](assets/release-channels.drawio.svg)
-
 The first use case will be modeled as the alpha channel. Modules available in the alpha channel are developed with all quality measures in place (functional correctness, security, etc.), but they might still have unstable API or be changed without keeping backward compatibility. When you use a module from the alpha channel, you won't get full SLA guarantees for that module or other modules that are affected (directly or indirectly).
 
 The second use case (deciding when updates should be applied) will require 2 production-grade channels with a different update schedule. The fast channel will get updates as soon as they are released and have passed all quality gates. The regular channel will get updates a few days later. Customers can switch the entire cluster or a particular component to the fast channel to check if the upstream changes do not cause any issues with their workload. Changing back to the regular channel is possible, but the module version won't be downgraded - the next version has to reach the channel to be applied.
 
-Hotfixes will be delivered to all channels immediately (TODO: how to apply a hotfix for the release that is not available in the current channel).
+Moving module versions between channels should be automated as much as possible using a promotion strategy suitable for the particular version change (major/minor/patch defined by [semver.org](https://semver.org/)) or priority (regular/hot-fix).
```

Comment:
Regarding this one, not a change request but rather a clarification and a sanity check on how it should be approached in the new architecture we are currently working on.

In my opinion, the promotion topic should be isolated from the Lifecycle Manager's logic. This will provide better separation and result in a clean cut between the responsibilities of the components. Therefore, my proposed solution would be:
- Have the SemVer kept for the Modules
- Make the Lifecycle Manager a bit channel-agnostic (we won't separate from that completely as channels are an essential part of current API implementation)

As for the result, Lifecycle Manager will [no longer use an enum for the release channel matching](https://github.com/kyma-project/lifecycle-manager/issues/324), it will just do the generic string comparison. This will provide a way of creating ad-hoc testing release channels in the dev environments (if needed) that can be used for some extreme-case upgrade scenarios that have to be tested (and by the way it will secure the implementation from any changes in the naming of the channels).

Then, the Lifecycle Manager will have 3 "primitive" scenarios to handle - installation, upgrade and downgrade. SemVer of the currently installed and the new Module would be used for determining if any action is needed, and as agreed - downgrades will not be supported.

And then - any automation taking care of the promotion logic of the modules can be implemented independently.

Reply:
```suggestion
Moving module versions between channels can be done with a simple pull request (bump version). Later on, it can be automated using a promotion strategy suitable for the particular version change (major/minor/patch defined by [semver.org](https://semver.org/)) or priority (regular/hot-fix), but it is not the highest priority for the first iteration.
```

Reply:
Yes - I agree. Promotion logic is not part of the lifecycle manager - it is CI/CD automation.